miio/gateway.py

```diff
@@ -75,21 +82,88 @@ class Plug(SubDevice):
     properties = ["power", "neutral_0"]
 
 
-class Gateway(Device):
+class GatewayAlarm(Device):
+    """Main class representing the Xiaomi Gateway Alarm."""
+
+    def __init__(self, Device) -> None:
```

Comment:
In principle I agree, I would think it makes sence to place this as a subdevice of the gateway, however because the gateway class is a mess at the moment, I wanted it to be seperated for the moment so I can only import the GatewayAlarm class and not import the gateway class

Reply:
So, in my opinion it should really belong underneath the Gateway, e.g., be exposed via `alarm()` returning the (sub)device. I'm just trying to think what's the sane way to expose the commands for the command line interface, but perhaps that can be figured out later.

So I'd say the following is the way to go:
1. Add `self._alarm` and initialize it with `GatewayAlarm(self)`
2. `GatewayAlarm` should not itself be a subclass of Device, but it should simply execute the commands over the reference. This will make it simple to interface with it as the relevant functionality is still contained outside the main class.
3. Add `def alarm()` returning the stored instance (maybe define it even as a `@property`, so that `gw_device.alarm.disarm()` would work).

When I do my cleanup for the rest of the parts, I'll try to organize them similarly (e.g., `Light` going under `GatewayLight` etc., to keep the main interface for `Gateway` clean(er).

What do you think? Importing `Gateway` itself won't cause any side effects, so it's just a matter of obtaining access to the alarm interface and ignoring the rest for the time being.

Reply:
@rytilahti I have reorganized the classes as you proposed